### DISCUSSION TITLE
Using magicbell with Deno and their Fresh framework

### DISCUSSION THREAD

**Discussion by mrkmadsn** [magicbell/community#190] [Questions] [answered]
Have anyone had succes with adding magicbell into a [Deno](https://deno.land) project?? . I am currently in the process of building a project inside the [Fresh ](https://fresh.deno.dev) framework, but having issue getting the packages to work properly, have tried both the javascript and react package solutions...

Thanks again

**Comment by smeijer** [comment]
Hi @MaDsEm88 ,

I don't have experience with Deno, but I'm happy to look into it. Do you have any more info on what doesn't work? Is there an error message / stack that you're able to share?

**Comment by mrkmadsn** [comment]
@smeijer , that would be awesome if you have the time. Could probably make it work in time, but as I have other things i also need to figure out, so it would be nice to just be able to implement it without issues.. ;)

But have created the files one with react and one with javascript where the errors has been described. When using deno and especially the fresh framework, all of us using it loves the structure but because it runs without the use of node_modules we have to implement packages and solution in other ways..

![codeimage-snippet_4 (1) (1)](https://github.com/magicbell-io/community/assets/91232612/fd0b9bc1-72da-4532-abbb-8f7bb02566fb)


![codeimage-snippet_4 (2) (1)](https://github.com/magicbell-io/community/assets/91232612/3c7400d6-f9f1-48a8-9ae4-d842210eb9cc)



And thanks again....

**Reply by smeijer** [reply, marked as answer]
The one using `@magicbell/embeddable` is due to server-side rendering. That SDK does not support server-side rendering, so you'd need to opt out of that for this one component. 

For the react SDK, I'm not sure why, but esm.sh seems to load `https://esm.sh/v135/fetch-addons@1.2.0/denonext/headers.js` instead of `https://esm.sh/v135/fetch-addons@1.2.0/denonext/dist/headers.js` (note the dist folder). Do you know why it would do that?

**Reply by mrkmadsn** [reply]
No idea why it would show that. Normally when we wrap the react package it will show if we need to add further packages to make it work or we can force it to show by using the * before we type the package name like "https://esm.sh/133/*packagename", that will show if the package is using other underlying packages that also needs to be wrapped., so have no idea why it would show https://esm.sh/v135/fetch-addons@1.2.0/denonext/headers.js
